File: java_solution/ReceiptFormatter.java
import java.util.*;

public class ReceiptFormatter {

  public static String formatReceipt(Receipt receipt) {
    String output = "";
    List<Item> items = receipt.getItems();

    for (Item item : items) {
      output += String.format(
        "%d%s %s: %.2f\n",
        item.getQuantity(),
        item.isImported() ? " imported" : "",
        item.getName(),
        item.getPrice()
      );
    }
    output += String.format(
      "Sales Taxes: %.2f\nTotal: %.2f",
      receipt.getTaxes(),
      receipt.getTotal()
    );
    return output;
  }
}

File: ruby_solution/receipt_formatter.rb
class ReceiptFormatter

  def self.format_receipt(receipt)
    i = 0
    output = ''
    while i < receipt.items_list.length
      output += "#{receipt.items_list[i].quantity}#{receipt.items_list[i].is_imported ? ' imported' : ''} #{receipt.items_list[i].name}: #{'%.02f' % receipt.items_list[i].price}\n"
      i += 1
    end
    output + "Sales Taxes: #{'%.02f' % receipt.calculate_taxes}\nTotal: #{'%.02f' % receipt.calculate_total}"
  end

end